hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/KeywordService.kt

```diff
@@ -0,0 +1,268 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services
+
+import com.google.gson.Gson
+import com.microsoft.applicationinsights.TelemetryClient
+import org.opensearch.action.search.SearchRequest
+import org.opensearch.action.search.SearchResponse
+import org.opensearch.common.unit.TimeValue
+import org.opensearch.index.query.BoolQueryBuilder
+import org.opensearch.index.query.MultiMatchQueryBuilder
+import org.opensearch.index.query.Operator
+import org.opensearch.index.query.QueryBuilder
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.common.config.OpenSearchIndexConfiguration.Companion.INDEX_ALIAS
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.search.config.AuthenticationHolder
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto.KeywordRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto.PaginationRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto.SearchType
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.exceptions.BadRequestException
+import java.util.concurrent.TimeUnit
+
+@Service
+class KeywordService(
+  private val elasticsearchClient: SearchClient,
+  private val gson: Gson,
+  private val telemetryClient: TelemetryClient,
+  private val authenticationHolder: AuthenticationHolder,
+  @Value("\${search.keyword.max-results}") private val maxSearchResults: Int = 200,
+  @Value("\${search.keyword.timeout-seconds}") private val searchTimeoutSeconds: Long = 10L,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  private fun validateKeywordRequest(keywordRequest: KeywordRequest) {
+    if (keywordRequest.prisonIds.isNullOrEmpty()) {
+      log.warn("Invalid keyword search  - no prisonIds specified to filter by")
+      throw BadRequestException("Invalid keyword search  - please provide prison locations to filter by")
+    }
+  }
+
+  fun findByKeyword(keywordRequest: KeywordRequest): Page<Prisoner> {
+    log.info("Received keyword request ${gson.toJson(keywordRequest)}")
+
+    validateKeywordRequest(keywordRequest)
+    val searchSourceBuilder = createSourceBuilder(keywordRequest)
+    val searchRequest = SearchRequest(arrayOf(INDEX_ALIAS), searchSourceBuilder)
```

Comment:
I'm gonna rename this - I now find it annoying - will change to PRISONER_INDEX or just PRISONER

Reply:
prefer `PRISONER_INDEX`